docs/proposals/cluster-templates.md

```diff
@@ -0,0 +1,47 @@
+## Introduction
+
+A cluster template is a reusable cluster template object. The Kubernetes CRD will be used for this purpose.
+A cluster template allows you to specify a provider, node layout, and configurations to materialize a cluster instance
+via Kubermatic API or UI.
+
+During cluster creation or after successful creation the user can store cluster definition as a template.
+
+### Scope
+
+The cluster templates should be accessible from different levels. Should have global (admin), project or user scope.
+Template management will be available from project level.
+We also need a central template management place for admins.
+
```

Comment:
Potential an enforcement of Templates should be respected as well

Reply:
Potentially it's possible but not so easy. I will prepare the whole implementation to support this in the future. The best will be to label the newly created clusters to indicate that they are created from the template